### PR TITLE
enable run vlm with pytorch engine in gradio

### PR DESCRIPTION
**Pull request by RunningLeon**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily receiving feedbacks. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

fix to enable run vlm with pytorch engine in gradio
image

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by RunningLeon** [comment]
lint error of markdown links will be fixed in pr https://github.com/InternLM/lmdeploy/pull/1931

**Comment by lvhan028** [comment]
May merge main to resolve lint issue

**Comment by RunningLeon** [comment]
FYI: tested on `gradio==4.36.1`

File: lmdeploy/serve/gradio/vl.py [hunk 1]
```diff
@@ -174,7 +174,10 @@ def cancel(chatbot, session):
 
     def reset(session):
         """Reset a new session."""
-        stop(session)
+        if session is None:
```
Comment (AllentDan) [inline lmdeploy/serve/gradio/vl.py:177]: Why is it a None?

Comment (RunningLeon) [reply lmdeploy/serve/gradio/vl.py:177]: press `reset` button before input anything, the `session is None`

**Review by AllentDan — APPROVED** [review]
LGTM

**Pull request merged**